### PR TITLE
Contributes 6 new device types for Nokia SR Linux product line

### PR DESCRIPTION
**Pull request by jbemmel**
* D1 (management switch)
* D2/D2L
* D3/D3L
* H3

All systems come in AC and DC variants, with airflow Front-2-Back or Back-2-Front. Now modeled as separate PSU/FAN modules with airflow attribute (to be discussed/added)

### CODE REVIEW

**Review by danner26 — CHANGES_REQUESTED** [review]
Hello, thank you for your contribution! Please review the requested changes and let me know if you have any questions. In the meantime I am going to mark this PR as a draft, and once you are ready you can mark it ready for review and ping me if desired.

File: device-types/Nokia/7220-IXR-H3.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+manufacturer: Nokia
+model: 7220 IXR-H3 32*400GE
+slug: 7220-ixr-h3 # As reported by cSRL in /platform/chassis/type
+part_number: 3HE16425AA
+u_height: 1
+is_full_depth: true  # 21.65"
```
[the remark this answers is not in the document]
Comment (jbemmel) [reply device-types/Nokia/7220-IXR-H3.yaml:7]: All devices come in 2 variants: Front-2-back and Back-2-front. Does that mean we would have to model each separately? (leading to x2 duplication of the same information)

Similar question for DC/AC variants, should we model those as separate device types?

[the remark this answers is not in the document]
Comment (danner26) [reply device-types/Nokia/7220-IXR-H3.yaml:7]: @jbemmel do the devices have different model numbers based on airflow direction? There are a handful of devices that are like this and indeed they are duplicated (besides airflow) with their specific model numbers being the differentiation

In regards to the AC/DC variants, again those PSUs appear to be modular. In this case the device should be modelled as is. Then in this file you make the PSUs module-bays, and create the PSUs as module-types as described in my other comments

File: device-types/Nokia/7220-IXR-D1.yaml [hunk 2]
```diff
@@ -0,0 +1,128 @@
+---
+manufacturer: Nokia
+model: 7220 IXR-D1 1/10GE
+slug: 7220-ixr-d1 # As reported by cSRL in /platform/chassis/type
+part_number: 3HE15621AA
+u_height: 1
+is_full_depth: false  # 15.75"
```
[the remark this answers is not in the document]
Comment (jbemmel) [reply device-types/Nokia/7220-IXR-D1.yaml:7]: added in modules

File: device-types/Nokia/7220-IXR-D2.yaml [hunk 3]
```diff
@@ -0,0 +1,138 @@
+---
+manufacturer: Nokia
+model: 7220 IXR-D2 25/100GE
+slug: 7220-ixr-d2 # As reported by cSRL in /platform/chassis/type
+part_number: 3HE15622AA
+u_height: 1
+is_full_depth: false  # 18.11"
```
[the remark this answers is not in the document]
Comment (jbemmel) [reply device-types/Nokia/7220-IXR-D2.yaml:7]: no, not available

[the remark this answers is not in the document]
Comment (jbemmel) [reply device-types/Nokia/7220-IXR-D2.yaml:7]: added in modules

File: device-types/Nokia/7220-IXR-D2L.yaml [hunk 4]
```diff
@@ -0,0 +1,142 @@
+---
+manufacturer: Nokia
+model: 7220 IXR-D2L 25/100GE
+slug: 7220-ixr-d2l # As reported by cSRL in /platform/chassis/type
+part_number: 3HE17645AA
+u_height: 1
+is_full_depth: true  # 21.10"
```
[the remark this answers is not in the document]
Comment (jbemmel) [reply device-types/Nokia/7220-IXR-D2L.yaml:7]: no, not available

File: device-types/Nokia/7220-IXR-D3.yaml [hunk 5]
```diff
@@ -0,0 +1,94 @@
+---
+manufacturer: Nokia
+model: 7220 IXR-D3 32*100GE
+slug: 7220-ixr-d3 # As reported by cSRL in /platform/chassis/type
+part_number: 3HE15623AA
+u_height: 1
+is_full_depth: false  # 18.11"
```
[the remark this answers is not in the document]
Comment (jbemmel) [reply device-types/Nokia/7220-IXR-D3.yaml:7]: no, not available

File: device-types/Nokia/7220-IXR-D3L.yaml [hunk 6]
```diff
@@ -0,0 +1,94 @@
+---
+manufacturer: Nokia
+model: 7220 IXR-D3L 32*100GE
+slug: 7220-ixr-d3l # As reported by cSRL in /platform/chassis/type
+part_number: 3HE17646AA
+u_height: 1
+is_full_depth: true  # 21.28"
```
[the remark this answers is not in the document]
Comment (jbemmel) [reply device-types/Nokia/7220-IXR-D3L.yaml:7]: no, not available

File: device-types/Nokia/7220-IXR-H3.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbemmel) [reply device-types/Nokia/7220-IXR-H3.yaml:7]: no, not available

[the remark this answers is not in the document]
Comment (jbemmel) [reply device-types/Nokia/7220-IXR-H3.yaml:7]: ok, modeled both PSUs and fans as modules

**Comment by jbemmel** [comment]
Looks like the system does not allow 'airflow' attribute for fan modules, but these modules do have a specific airflow direction - it is essential to their existence

Could we update Netbox to allow airflow attributes for modules?

**Comment by danner26** [comment]
Yup I agree, for the time being to get this PR moving, could you just create the modules without airflow? You could also add a comment with the airflow direction so that once this is possible we can retroactively go back and fix these

File: device-types/Nokia/7220-IXR-D2L.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (jbemmel) [reply device-types/Nokia/7220-IXR-D2L.yaml:7]: as discussed, airflow is specific to the PSU/fan modules

**Comment by danner26** [comment]
Hey @jbemmel I would recommend adding `position` to the module-types. When a module referenced `{module}` it will use what is in position
For example if your module is `- name: PSU{module}` and you install it to a slot with `position: 3`, you will end up with  `PSU3`

Let me know if you need help with this. If you, please send me an invite to your fork

**Comment by danner26** [comment]
Just a reminder the positions need to be strings, so you need to write them as `'1'`

**Review by danner26 — APPROVED** [review]

**Pull request merged**